### PR TITLE
[#187364679]User should be able to login via google 

### PR DESCRIPTION
**Pull request by mpanoathos**
## What does this PR do?
This PR includes google authentication: User will be able login via his/er google account and will be redirected to the specific page according to his/er user role
-Buyer : Login Page
-Seller: verify otp page
-Admin: admin dashboard
## Type of change

- [x] New feature (non-breaking change which adds functionality)

## How Has This Been Tested?
-On the PR preview link
## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

## What are the relevant pivotal tracker/Trello stories?
[#187364679]

### CODE REVIEW

File: src/test/views/Login.test.tsx [hunk 1]
```diff
@@ -16,7 +16,7 @@ import "@testing-library/jest-dom";
 import axios from "axios";
 import { BrowserRouter } from "react-router-dom";
 import TextInput from "../../components/TextInput";
-
+// import { loginWithGoogle } from '../../views/Login';
```
Comment (AngeloChristian1) [inline src/test/views/Login.test.tsx]: You can remove this line if it isn't being used

File: src/views/Login.tsx [hunk 2]
```diff
@@ -21,7 +22,25 @@ const Login: React.FC = () => {
...
+      window.open(`${backendUrl}/users/google`, "_self");
+
+      const urlParams = new URLSearchParams(window.location.search);
+      const token = urlParams.get('token');
 
+      if (token) {
+        const bearerToken = `Bearer ${token}`;
+        localStorage.setItem('AUTH_TOKEN', bearerToken);
+        dispatch(setIsLoggedIn(true));
+      }
```
Comment (AngeloChristian1) [inline src/views/Login.tsx:39]: Remember to redire t back to home depending on user

**Review by AngeloChristian1 — COMMENTED** [review]
Good job. Keep it up 
You can work on the rest

File: src/views/Login.tsx (hunk 2 above)
Comment (mpanoathos) [reply src/views/Login.tsx:39]: Yes it was done on the backend

**Review by AngeloChristian1 — APPROVED** [review]

**Review by EricNiyo73 — APPROVED** [review]

**Review by munezerobagira — APPROVED** [review]

**Pull request merged**